### PR TITLE
Adding loader for FSDnoisy18K

### PR DESCRIPTION
**Pull request by genisplaja**
### Description

Please include the following information at the top level docstring for the dataset's module mydataset.py:

- [x] Describe annotations included in the dataset
- [x] Indicate the total duration of the dataset in hours, and (optionally) also list the number of individual files
- [x] Mention the origin of the dataset (e.g. creator, institution)
- [x] Describe the type of audio included in the dataset
- [x] Indicate any relevant papers related to the dataset

#### Dataset loaders checklist:

- [x] Create a script in `scripts/`, e.g. `make_my_dataset_index.py`, which generates an index file.
- [x] Run the script on the canonical version of the dataset and save the index in `soundata/indexes/` e.g. `my_dataset_index.json`.
- [x] Create a module in soundata, e.g. `soundata/my_dataset.py`
- [x] Create tests for your loader in `tests/`, e.g. `test_my_dataset.py`
- [x] Add your module to `docs/source/soundata.rst` and `docs/source/quick_reference.rst`
- [x] Run `tests/test_full_dataset.py` on your dataset.

I am also removing a couple of unnecessary lines I forgot to take out from the FSD50K loader.

### CODE REVIEW

File: soundata/datasets/fsdnoisy18k.py [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+    Attributes:
+        audio_path (str): path to the audio file
+        clip_id (str): clip id
+
+    Properties:
+        tag (soundata.annotations.Tags): tag (label) of the clip + confidence
+        aso_id (str): the id of the corresponding category as per the AudioSet Ontology
+        manually_verified (str): flag to indicate whether the clip belongs to the clean portion (1), or to the noisy portion (0) of the train set
+        noisy_small (str): flag to indicate whether the clip belongs to the noisy_small portion (1) of the train set
+        split (str): flag to indicate whether the clip belongs the train or test split
```
Comment (magdalenafuentes) [inline soundata/datasets/fsdnoisy18k.py]: Hey @genisplaja the description of these properties is making me think they are binary indicators, like `boolean` flags. Is it the case or are they labels e.g. "clean_portion" or "noisy_portion"?

File: soundata/datasets/fsdnoisy18k.py [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+        metadata_test_path = os.path.join(
+            self.data_home, "FSDnoisy18k.meta", "test.csv"
+        )
+
+        if not os.path.exists(metadata_train_path):
+            raise FileNotFoundError(
+                "Train metadata not found. Did you run .download()?"
+            )
+        if not os.path.exists(metadata_test_path):
+            raise FileNotFoundError("Test metadata not found. Did you run .download()?")
```
Comment (magdalenafuentes) [inline soundata/datasets/fsdnoisy18k.py:293]: Line not covered by tests

File: soundata/datasets/fsdnoisy18k.py [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+        with open(metadata_train_path, "r") as f:
+            reader = csv.reader(f, delimiter=",")
+            next(reader)
+            for row in reader:
+                metadata_index[row[0].replace(".wav", "")] = {
+                    "split": "train",
+                    "tag": row[1],
+                    "aso_id": str(row[2]),
+                    "manually_verified": int(row[3]),
+                    "noisy_small": int(row[4]),
```
Comment (magdalenafuentes) [inline soundata/datasets/fsdnoisy18k.py:306]: Note that this appears as `str` in the `readme` of the `clip`

File: tests/datasets/test_fsdnoisy18k.py [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+        "audio_path": "tests/resources/sound_datasets/fsdnoisy18k/FSDnoisy18k.audio_test/564.wav",
+        "clip_id": "564",
+    }
+
+    expected_property_types_test = {
+        "audio": tuple,
+        "tags": annotations.Tags,
+        "split": str,
+        "aso_id": str,
+        "manually_verified": None.__class__,
```
Comment (magdalenafuentes) [inline tests/datasets/test_fsdnoisy18k.py]: Do you need the `__class__` here?

File: tests/datasets/test_fsdnoisy18k.py [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+
+def test_load_audio():
+    dataset = fsdnoisy18k.Dataset(TEST_DATA_HOME)
+    clip = dataset.clip("17")
+    audio_path = clip.audio_path
+    audio, sr = fsdnoisy18k.load_audio(audio_path)
+    assert sr == 44100
+    assert type(audio) is np.ndarray
+    assert len(audio.shape) == 1  # check audio is loaded as mono
+    assert len(audio) == 47786
```
Comment (magdalenafuentes) [inline tests/datasets/test_fsdnoisy18k.py:59]: Not one sec but small enough :P

File: tests/datasets/test_fsdnoisy18k.py [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+    # Testing metadata from a training clip
+    default_clipid = "17"
+    dataset = fsdnoisy18k.Dataset(TEST_DATA_HOME)
+    clip = dataset.clip(default_clipid)
+    clip_metadata = clip._metadata()[default_clipid]
+
+    assert clip_metadata["tag"] == "Walk_or_footsteps"
+    assert clip_metadata["aso_id"] == "/m/07pbtc8"
+    assert clip_metadata.get("manually_verified") == 1
+    assert clip_metadata.get("noisy_small") == 0
```
Comment (magdalenafuentes) [inline tests/datasets/test_fsdnoisy18k.py:107]: Getting back to my comment in the loader code, I think this should probably be `boolean`...

**Review by magdalenafuentes — COMMENTED** [review]
@genisplaja this is looking great :)

Just left a couple minor comments and will be ready to go!

File: soundata/datasets/fsdnoisy18k.py (hunk 1 above)
Comment (genisplaja) [reply soundata/datasets/fsdnoisy18k.py]: No, they are definitely a bool (whether 1 or 0). My bad... I copy pasted the lines and forgot to update the data type. I am using 1 and 0 here to preserve the dataset nomeclature. Is that OK?

File: tests/datasets/test_fsdnoisy18k.py (hunk 4 above)
Comment (genisplaja) [reply tests/datasets/test_fsdnoisy18k.py]: If I test it without the `__class__` it does not work! For that reason I am using it. Do you know if there's a different way to test that an attribute is `None` here? Thanks!

File: soundata/datasets/fsdnoisy18k.py (hunk 1 above)
Comment (magdalenafuentes) [reply soundata/datasets/fsdnoisy18k.py]: Yes it's ok, if they are `int` or `float` (I meant no `str` XD) that should work as `boolean`

File: tests/datasets/test_fsdnoisy18k.py (hunk 4 above)
Comment (magdalenafuentes) [reply tests/datasets/test_fsdnoisy18k.py]: I see, you're right. It's due to the check we do in `test_utils` (` assert isinstance(getattr(clip, prop), expected_property_types[prop]`). 

we could change that line to check for `None` or maybe change `None.__class__` for `type(None)` which might be clearer? (the result is the same though). I'm ok either way

**Review by magdalenafuentes — APPROVED** [review]
@genisplaja see my comments and feel free to merge anytime!

File: tests/datasets/test_fsdnoisy18k.py (hunk 4 above)
Comment (genisplaja) [reply tests/datasets/test_fsdnoisy18k.py]: Agree! Le's use `type(None)` it is way clearer! Thanks for the suggestion @magdalenafuentes. I guess the loader is ready to go now:)

File: soundata/datasets/fsdnoisy18k.py [hunk 7]
```diff
@@ -0,0 +1,319 @@
...
+        metadata_test_path = os.path.join(
+            self.data_home, "FSDnoisy18k.meta", "test.csv"
+        )
+
+        if not os.path.exists(metadata_train_path):
+            raise FileNotFoundError(
+                "Train metadata not found. Did you run .download()?"
+            )
+        if not os.path.exists(metadata_test_path):
+            raise FileNotFoundError("Test metadata not found. Did you run .download()?")
```
Comment (magdalenafuentes) [inline soundata/datasets/fsdnoisy18k.py:293]: Just this line is not covered by tests, let's just fix that and ready to merge! <3

**Review by magdalenafuentes — APPROVED** [review]

**Comment by genisplaja** [comment]
@magdalenafuentes all tested:) I had to add tiny download resources to get all the lines tested. I guess it's alright! If so, the loader is ready to go! Sorry I didn't notice there was a line missing.

**Comment by magdalenafuentes** [comment]
No worries at all! Merging :) Great job!

**Pull request merged**